alchemist/alchemist-smartcam/src/main/kotlin/it/unibo/alchemist/model/smartcam/VisibleNodeImpl.kt

```diff
@@ -0,0 +1,25 @@
+package it.unibo.alchemist.model.smartcam
+
+import it.unibo.alchemist.model.interfaces.Node
+import it.unibo.alchemist.model.interfaces.Position
+import it.unibo.alchemist.model.interfaces.VisibleNode
+
+/**
+ * Basic implementation of [VisibleNode]
+ */
+class VisibleNodeImpl<T, P : Position<P>>(
+    override val node: Node<T>,
+    override val position: P
+) : VisibleNode<T, P> {
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
This is a one-liner:
`= other is VisibleNodeImpl && other.node == node`

Reply:
Done